packages/medusa/src/strategies/batch-jobs/order/export.ts

```diff
@@ -0,0 +1,263 @@
+    req: Express.Request
+  ): Promise<AdminPostBatchesReq> {
+    const {
+      limit,
+      offset,
+      order,
+      fields,
+      expand,
+      filterable_fields,
+      ...context
+    } = batchJob.context as OrderExportBatchJobContext
+
+    const listConfig = prepareListQuery(
+      {
+        limit,
+        offset,
+        order,
+        fields,
+        expand,
+      },
+      {
+        isList: true,
+        defaultRelations: this.defaultRelations_,
+        defaultFields: this.defaultFields_,
+      }
+    )
+
+    batchJob.context = {
+      ...(context ?? {}),
+      list_config: listConfig,
+      filterable_fields,
+    }
+
+    return batchJob
+  }
+
+  async processJob(batchJobId: string): Promise<void> {
+    let offset = 0
+    let limit = this.BATCH_SIZE
+    let advancementCount = 0
+    let orderCount = 0
+
+    return await this.atomicPhase_(
+      async (transactionManager) => {
+        let batchJob = (await this.batchJobService_
+          .withTransaction(transactionManager)
+          .retrieve(batchJobId)) as OrderExportBatchJob
+
+        const { writeStream, fileKey, promise } = await this.fileService_
+          .withTransaction(transactionManager)
+          .getUploadStreamDescriptor({
+            name: "order-export",
+            ext: "csv",
+          })
+
+        advancementCount =
+          batchJob.result?.advancement_count ?? advancementCount
+        offset = (batchJob.context?.list_config?.skip ?? 0) + advancementCount
+        limit = batchJob.context?.list_config?.take ?? limit
+
+        const { list_config = {}, filterable_fields = {} } = batchJob.context
+        const [, count] = await this.orderService_.listAndCount(
+          filterable_fields,
+          {
+            ...list_config,
+            order: { created_at: "DESC" },
+            skip: offset,
+            take: Math.min(batchJob.context.batch_size ?? Infinity, limit),
+          }
+        )
+
+        const lineDescriptor = this.getLineDescriptor(
+          list_config.select as string[],
+          list_config.relations as string[]
+        )
+
+        const header = this.buildHeader(lineDescriptor)
+        writeStream.write(header)
+
+        orderCount = batchJob.context?.batch_size ?? count
+
+        const result = batchJob.result
+        let orders = []
+
+        while (offset < orderCount) {
+          orders = await this.orderService_
+            .withTransaction(transactionManager)
+            .list(filterable_fields, {
+              ...list_config,
+              skip: offset,
+              take: Math.min(orderCount - offset, limit),
+            })
+
+          orders.forEach(async (order) => {
+            writeStream.write(await this.buildCSVLine(order, lineDescriptor))
+          })
+
+          batchJob = (await this.batchJobService_
+            .withTransaction(transactionManager)
+            .update(batchJobId, {
+              result: {
+                ...result,
```

Comment:
question: The batch job service update method already spread the existing object (result, context) for us. either we should remove the result spreading from here or remove it from the update method and say that we always push the entire object. wdyt?